docs/extensibility/extend-rest-api-update-cart.md

```diff
@@ -0,0 +1,90 @@
+# Updating the cart with the Store API
+If you're an extension developer, and your extension does some server-side processing as a result of some client-side
+input, then you may want to use the `cart/extensions` endpoint to signal to
+the Store API that some processing needs to take place.
+
+Extensions may register a callback to run when the `cart/extensions` endpoint is hit. When it is, the server will
+execute all registered callbacks for a specified namespace.
+
+## Example use case
+You are the author of an extension that automatically applies valid coupons to a user's cart when they press a
+button.
+
+Your extension adds a button to the sidebar in the Cart and Checkout blocks using the [`DiscountsMeta`](./available-slot-fills.md) Slot.
+
+When this button is clicked, you want the server to do some work to figure out which coupons the customer is eligible
+for, and then apply them to the cart.
+
+When this is done, you expect to see the updated cart data in the client.
+
+## Registering a callback to run when the `cart/extensions` endpoint is hit
+Much like adding data to the Store API (described in more detail in
+[Exposing your data in the Store API](./extend-rest-api-add-data.md).) you may add the callback
+by invoking the `register_update_callback` method on the `ExtendRestApi` class from WooCommerce Blocks.
+
+```PHP
+
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\Domain\Services\ExtendRestApi;
+
+add_action('woocommerce_blocks_loaded', function() {
+ // ExtendRestApi is stored in the container as a shared instance between the API and consumers.
+ // You shouldn't initiate your own ExtendRestApi instance using `new ExtendRestApi` but should always use the shared instance from the Package dependency injection container.
+ $extend = Package::container()->get( ExtendRestApi::class );
+ $extend->register_update_callback(
+   [
+    'namespace' => 'my-extensions-unique-namespace',
+    'callback'  => /* Add your callable here */
+   ]
+ )
+} );
+```
```

Comment:
This would fit very well in the basic usage section as a sub item.

Reply:
Fixed by the new format.